From d6ef4a733b6c0f0cd62de3be3e4be603d401de3a Mon Sep 17 00:00:00 2001
Message-ID: <d6ef4a733b6c0f0cd62de3be3e4be603d401de3a.1707646257.git.sharlatanus@gmail.com>
In-Reply-To: <cover.1707646257.git.sharlatanus@gmail.com>
References: <cover.1707646257.git.sharlatanus@gmail.com>
From: Sharlatan Hellseher <sharlatanus@gmail.com>
Date: Sat, 10 Feb 2024 16:29:50 +0000
Subject: [PATCH 17/30] gnu: go-golang-org-x-mod: Move to (gnu packages-build).

* gnu/packages/golang.scm (go-golang-org-x-mod): Move from here ...
* gnu/packages/golang-build.scm: ... to here.

* gnu/packages/shellutils.scm: Add (gnu packages golang-build) module.

Change-Id: I88ce7ac1e264cb545a272f9c8a485747e84f7c82
---
 gnu/packages/golang-build.scm | 36 +++++++++++++++++++++++++++++++++++
 gnu/packages/golang.scm       | 34 ---------------------------------
 gnu/packages/shellutils.scm   |  1 +
 3 files changed, 37 insertions(+), 34 deletions(-)

diff --git a/gnu/packages/golang-build.scm b/gnu/packages/golang-build.scm
index ffedc946c2..a8e97e7acb 100644
--- a/gnu/packages/golang-build.scm
+++ b/gnu/packages/golang-build.scm
@@ -1,5 +1,7 @@
 ;;; GNU Guix --- Functional package management for GNU
 ;;; Copyright © 2020 Leo Famulari <leo@famulari.name>
+;;; Copyright © 2021 Ludovic Courtès <ludo@gnu.org>
+;;; Copyright © 2021 hackeryarn <artemchernyak@gmail.com>
 ;;; Copyright © 2023 Artyom V. Poptsov <poptsov.artyom@gmail.com>
 ;;; Copyright © 2023 Felix Lechner <felix.lechner@lease-up.com>
 ;;; Copyright © 2023 Katherine Cox-Buday <cox.katherine.e@gmail.com>
@@ -67,6 +69,40 @@ (define-public go-golang-org-x-exp
 @code{old} directory) packages.")
     (license license:bsd-3)))
 
+(define-public go-golang-org-x-mod
+  (let ((commit "7c05a442b7c1d1a107879b4a090bb5a38d3774a1")
+        (revision "0"))
+    (package
+      (name "go-golang-org-x-mod")
+      (version (git-version "0.7.0" revision commit))
+      (source
+       (origin
+         (method git-fetch)
+         (uri (git-reference
+               (url "https://github.com/golang/mod")
+               (commit commit)))
+         (file-name (git-file-name name version))
+         (sha256
+          (base32 "14r24fq3kn84k2y2jvvg8hwpy52a3q429pimrdwl5zwknbr2awmh"))))
+      (build-system go-build-system)
+      (arguments
+       '(#:import-path "golang.org/x/mod/"
+         #:tests? #f
+         #:phases (modify-phases %standard-phases
+                    ;; Source-only package
+                    (delete 'build))))
+      (home-page "https://golang.org/x/mod")
+      (synopsis "Tools to work directly with Go module mechanics")
+      (description
+       "This repository holds packages for writing tools that work directly
+with Go module mechanics.  That is, it is for direct manipulation of Go
+modules themselves.
+
+The specific case of loading packages should still be done by invoking the
+@command{go} command, which remains the single point of truth for package
+loading algorithms.")
+      (license license:bsd-3))))
+
 (define-public go-golang-org-x-net
   (let ((commit "8e0e7d8d38f2b6d21d742845570dde2902d06a1d")
         (revision "0"))
diff --git a/gnu/packages/golang.scm b/gnu/packages/golang.scm
index dd0ac7e3e8..e7b98028a5 100644
--- a/gnu/packages/golang.scm
+++ b/gnu/packages/golang.scm
@@ -3699,40 +3699,6 @@ (define-public go-golang-org-x-xerrors
       (home-page "https://godoc.org/golang.org/x/xerrors")
       (license license:bsd-3))))
 
-(define-public go-golang-org-x-mod
-  (let ((commit "7c05a442b7c1d1a107879b4a090bb5a38d3774a1")
-        (revision "0"))
-    (package
-      (name "go-golang-org-x-mod")
-      (version (git-version "0.7.0" revision commit))
-      (source (origin
-                (method git-fetch)
-                (uri (git-reference
-                      (url "https://github.com/golang/mod")
-                      (commit commit)))
-                (file-name (git-file-name name version))
-                (sha256
-                 (base32
-                  "14r24fq3kn84k2y2jvvg8hwpy52a3q429pimrdwl5zwknbr2awmh"))))
-      (build-system go-build-system)
-      (arguments
-       '(#:import-path "golang.org/x/mod/"
-         #:tests? #f
-         #:phases (modify-phases %standard-phases
-                    ;; Source-only package
-                    (delete 'build))))
-      (home-page "https://golang.org/x/mod")
-      (synopsis "Tools to work directly with Go module mechanics")
-      (description
-       "This repository holds packages for writing tools that work directly
-with Go module mechanics.  That is, it is for direct manipulation of Go modules
-themselves.
-
-The specific case of loading packages should still be done by invoking the
-@command{go} command, which remains the single point of truth for package
-loading algorithms.")
-      (license license:bsd-3))))
-
 (define-public go-github-com-hashicorp-go-uuid
   (package
     (name "go-github-com-hashicorp-go-uuid")
diff --git a/gnu/packages/shellutils.scm b/gnu/packages/shellutils.scm
index 48c5250c8c..e58e9e61fc 100644
--- a/gnu/packages/shellutils.scm
+++ b/gnu/packages/shellutils.scm
@@ -51,6 +51,7 @@ (define-module (gnu packages shellutils)
   #:use-module (gnu packages check)
   #:use-module (gnu packages flex)
   #:use-module (gnu packages golang)
+  #:use-module (gnu packages golang-build)
   #:use-module (gnu packages golang-xyz)
   #:use-module (gnu packages libunistring)
   #:use-module (gnu packages ncurses)
-- 
2.41.0

